core/src/main/java/io/grpc/internal/ServerCallImpl.java

```diff
@@ -209,7 +210,11 @@ public void close(Status status, Metadata trailers) {
   }
 
   private void closeInternal(Status status, Metadata trailers) {
-    checkState(!closeCalled, "call already closed");
+    if (closeCalled) {
```

Comment:
No. This was on purpose and is enforcing the API. The API says you can only close once. The problem is that our code is calling it when it shouldn't.